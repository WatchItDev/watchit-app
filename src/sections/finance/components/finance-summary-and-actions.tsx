import React from 'react';
import Stack from '@mui/material/Stack';
import FinanceWidgetSummary from '@src/sections/finance/components/finance-widget-summary';
import FinanceQuickTransfer from '@src/sections/finance/components/finance-quick-transfer';
import FinanceEarnTokens from '@src/sections/finance/components/finance-earn-tokens';
import { SummaryAndActionsProps } from '@src/sections/finance/types.ts';
import { useResponsive } from '@src/hooks/use-responsive.ts';

export const SummaryAndActions: React.FC<SummaryAndActionsProps> = (props) => {
  const {
    percent,
    widgetSeriesData,
    balanceFromRedux,
    following,
    loadingProfiles,
  } = props;
  const lgUp = useResponsive('up', 'lg');
  const mdUp = useResponsive('up', 'md');

  return (
    <Stack direction={{ lg: 'column', xlg: 'row' }} spacing={{ xs: 2, lg: 2 }}>
      <FinanceWidgetSummary
        title="Balance"
        color="primary"
        icon={
          percent > 0
            ? 'eva:diagonal-arrow-right-up-fill'
            : 'eva:diagonal-arrow-left-down-fill'
        }
        percent={percent}
        total={balanceFromRedux}
        chart={{ series: widgetSeriesData }}
      />
      {!mdUp && <FinanceQuickTransfer list={following} loading={loadingProfiles} />}
      {lgUp && <FinanceEarnTokens lgUp={lgUp} />}
    </Stack>
  );
};
