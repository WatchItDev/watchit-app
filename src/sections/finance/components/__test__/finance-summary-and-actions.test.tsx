import { describe, expect, it, vi } from "vitest";
import { render } from "@testing-library/react";

vi.mock("@web3auth/auth", () => ({
  keccak256Padded: vi.fn(),
  mimcGetConstants: vi.fn(),
}));

vi.mock("ethereum-cryptography/utils.js", () => ({
  keccak256: vi.fn(() => new Uint8Array()),
}));

vi.mock("@src/sections/finance/components/finance-widget-summary", () => ({
  default: () => <div>Mocked FinanceWidgetSummary</div>,
}));
vi.mock("@src/sections/finance/components/finance-quick-transfer", () => ({
  default: () => <div>Mocked FinanceQuickTransfer</div>,
}));
vi.mock("@src/sections/finance/components/finance-earn-tokens", () => ({
  default: () => <div>Mocked FinanceEarnTokens</div>,
}));

vi.mock("@src/hooks/use-responsive.ts", () => ({
  useResponsive: vi.fn(() => true),
}));

import { SummaryAndActions } from "../finance-summary-and-actions";

const mockProps = {
  percent: 0,
  widgetSeriesData: [],
  balanceFromRedux: 0,
  following: undefined,
  loadingProfiles: false,
};

const renderComponent = (props = mockProps) => render(<SummaryAndActions {...props} />);

describe("[COMPONENTS] <FinanceSummaryAndActions/>", () => {
  it("should match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
