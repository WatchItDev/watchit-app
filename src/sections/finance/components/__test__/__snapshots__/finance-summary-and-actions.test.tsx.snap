// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[COMPONENTS] <FinanceSummaryAndActions/> > should match snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1noynt0-MuiStack-root"
  >
    <div>
      Mocked FinanceWidgetSummary
    </div>
    <div>
      Mocked FinanceEarnTokens
    </div>
  </div>
</div>
`;
